### ISSUE TITLE
Confused about custom route paths

### ISSUE THREAD

**Issue by Flowko** [nuxt-modules/i18n#1917]
### Environment

- Operating System: `Windows_NT`
- Node Version:     `v18.14.2`
- Nuxt Version:     `3.2.3`
- Nitro Version:    `2.2.3`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `build`, `modules`, `css`, `typescript`, `image`, `i18n`
- Runtime Modules:  `@nuxtjs/tailwindcss@6.4.1`, `@nuxt/image-edge@1.0.0-27968280.9739e4d`, `nuxt-headlessui@1.1.1`, `@vueuse/nuxt@9.13.0`, `nuxt-icon@0.3.2`, `@nuxtjs/i18n@8.0.0-beta.10`, `@nuxtjs/device@3.1.0`, `nuxt-swiper@1.0.1`, `nuxt-og-image@1.4.22`
- Build Modules:    `-`

### Reproduction

sorry if this might be something that obvious, but im trying to understand why this doesnt work:
- have a page setup like this:
```
-pages
--shop
---[query].vue
```
inside that file, im just showing route object.
- inside of nuxt.config.ts
```ts
hooks: {
    "pages:extend"(pages) {
      pages.push({
        name: "engagement-rings-solitaire",
        path: "/shop/category=1&style=1",
        file: resolve(__dirname, "pages/shop/[query].vue"),
      });
    },
  },
i18n: {
    locales: [
      {
        code: "en",
        iso: "en-US",
        name: "English",
        file: "en-US.json",
      },
      {
        code: "fr",
        iso: "fr-FR",
        name: "Français",
        file: "fr-FR.json",
      },
    ],
    lazy: true,
    langDir: "lang",
    defaultLocale: "en",
    strategy: "prefix_except_default",
    customRoutes: "config",
    pages: {
      "shop/category=1&shape=2": {
        en: "/engagement-rings/solitaire",
        fr: "/bague-de-fiancaille/solitaire",
      },
    },
  },

```

### Describe the bug

so my initial thought that when i try to go to `/engagement-rings/solitaire` it should internaly load the shop page with `category=1&shape=2`, but it just loads a 404 page, and i cant find a way to get arround that.

if what im doing is wrong, please can u share a solution to get it working. thanks

also it seems that i get a warn:
[@nuxtjs/i18n]: Couldn't find AnalizedNuxtPageMeta by NuxtPage (/shop/category=1&shape=2), so no custom route for it

**Comment by Flowko** [comment]
i believe this might be related as well, but not sure #1837

**Comment by memic84** [comment]
Is your project in a monorepo, or using the Nuxt extends functionality? Could the problem be that `pages:extend` is executed after the i18n module is loaded, or something else wrong in the loading order?

**Comment by Flowko** [comment]
not a monorepo, and like u can see above, im using the `pages:extend` inside the hooks, inside of nuxt.config.ts
```js
hooks: {
    "pages:extend"(pages) {
      pages.push({
        name: "engagement-rings-solitaire",
        path: "/shop/category=1&style=1",
        file: resolve(__dirname, "pages/shop/[query].vue"),
      });
    },
  },
```

tbh im not sure if its extending routes after the i18 module is loaded, i'll have to check that

**Comment by Flowko** [comment]
@memic84 yep, it seems like pages:extend gets called after the i18n module has already loaded, i'll try to move my hook into a module that i can load before the i18n, hopefully that works

**Comment by memic84** [comment]
I'm am not sure what you are trying to accomplish with the hook pages:extend. 

If you are using the Pages option of this module, i believe that you need to use the name and a valid url format:

```
pages: {
      "shop/category=1&shape=2": {
        en: "/engagement-rings/solitaire",
        fr: "/bague-de-fiancaille/solitaire",
      },
    },
```  

The `shop/category=1&shape=2` seems invalid because you are using query params with the question mark sign.

```
pages: {
      "shop/[query]": {
        en: "/shop/:query",
        fr: "/boutique/:query",
      },
    },
```

**Comment by memic84** [comment]
I don't think that you should manually extend the pages that way. There is a issue with lokalized routes in this module, where if you use the layers (extend functionality), the extended directory is not used.

**Comment by Flowko** [comment]
i see, let me rephrase what im trying to achieve here:
- i want `fr/shop/bague-de-fiancailles` and `/shop/engagement-rings` to both be linked with a path that has, lets say i changed that [query].vue to [slug].vue, i want them to load the page with `slug=1`, is that possible to do ?

**Comment by memic84** [comment]
You should have something like this, which you have already i think: `/pages/shop/[slug].vue`

If you have different urls for other languages, than you could do handle this in the Pages option:

```
  i18n: {
    pages: {
      'shop/[slug]': {
        'en': '/shop/[slug]',
        'fr': '/boutique/[slug]' // if all the url's are the same i don't see a reason to change this in nuxt.config i18n/pages
      }
    }
  }
```

I assume that you are fetching dynamic data for your slug, so that is always filled by a menu, or other content items on the page. The slug (route param) that you get in [slug].vue, is used to fetch the correct data.

Otherwise could you create a reproduction on stackbilitz.

**Comment by Flowko** [comment]
yeah i see, i guess my only option is to get whatever `slug` they insert and return the correct data for it by checking what products are linked with it, but yeah i was hoping there's an option so i wouldn't have to add another layer to check for slugs, instead i get like the category id directly and return whatever data is needed, but thanks for ur help!

**Comment by memic84** [comment]
You could also use id's, like this /shop/[id] and but then you would have url's like this /shop/1 instead of SEO nice urls. 

If you are building with SSR, one thing that we did, if we needed to to multiple calls, lets say:
1) fetch product id by slug name
2) fetch product by id

We put this in server/api routes, and did all the data fetching part in node instead at the client or pages.
